### ISSUE TITLE
Performance improvement suggestions?

### ISSUE THREAD

**Issue by ritwickdsouza** [kolofordjango/kolo#23]
Hey all, kudos to building kolo. Looks like a solid debugger.
My only problem with it now that makes it unusable in our app is the serious performance hit that makes the app unusable.

Do you have any suggestions for tweaks I could make in the settings to improve the performance?

**Comment by LilyFoote** [comment]
Hi @ritwickdsouza.

The one thing that comes to mind is to check whether you have Kolo and Django Debug Toolbar running at the same time. This is a known case of a significant slowdown.

Otherwise, we'll need to know more about your app and how you're using Kolo to give useful suggestions. Feel free to discuss here, [on our discord](https://discord.com/channels/899363528660635738/899363528660635742) or [in a video call with @wilhelmklopp](https://calendly.com/wilhelmklopp).

**Comment by valentinogagliardi** [comment]
Had a similar problem, I was running both Kolo and DDT. Disabling the debug toolbar solved the issue.

**Comment by caleb15** [comment]
I also have a performance issue, even with the slow panels of debug toolbar disabled. We have a large (~5k files) codebase, so it's understandably slow for Kolo to go through.

I'm using the following panels, none of which should cause a significant slowdown:
```python
    DEBUG_TOOLBAR_PANELS = (
        'debug_toolbar.panels.timer.TimerPanel',
        'debug_toolbar.panels.cache.CachePanel',
        'debug_toolbar.panels.signals.SignalsPanel',
        'debug_toolbar.panels.logging.LoggingPanel',
        'debug_toolbar.panels.redirects.RedirectsPanel',
    )
```

**Comment by caleb15** [comment]
Talked about this in a call, we still see a performance hit even with debug toolbar disabled completely. One option I forgot to mention during the call is that you could have a optional mode to disable variable storage in the frames. My guess is skipping serialization and storage of variables could save a significant amount of time. 

Even without seeing the variables, the visualization, SQL requests, and all those other goodies could still be useful.

**Comment by wilhelmklopp** [comment]
Hey everyone! Appreciate all the great ideas here ❤️ 

Just wanted to briefly mention that we've shipped a number of performance improvements, most notably in in 2.1.0 (released 2022-10-27) and 2.3.0 (released 2022-12-01).

Most of these performance improvements don't require any setting changes and you should just generally notice things being faster. 


But there is one improvement which we're keeping as opt-in and behind a flag for now: `use_rust = true`

@LilyFoote has started converting bits of kolo to rust for improved performance and we've already seeing quite promising results in our internal testing 🎉 

To use the faster rust implementation set use_rust = true in `.kolo/config.toml`. Your kolo config file might end up looking like this:

```toml
use_rust = true

[filters]

ignore_request_paths = ["/static/", "/favicon.ico"]
ignore_frames = ["main/middleware.py"]
```
We're still at the start of the rust journey and expect many more improvements to come in the future, but wanted to share this with you all since we're already seeing performance improvements with what we have today :)
